sdks/java/io/synthetic/src/test/java/org/apache/beam/sdk/io/synthetic/SideInputLoadIT.java

```diff
@@ -0,0 +1,188 @@
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionView;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Performance fanout test for {@link CoGroupByKey}. */
+@RunWith(JUnit4.class)
+public class SideInputLoadIT {
+
+  private static Options options;
+
+  private static SyntheticBoundedIO.SyntheticSourceOptions inputOptions;
+
+  private static SyntheticBoundedIO.SyntheticSourceOptions sideInputOptions;
+
+  @Rule public TestPipeline pipeline = TestPipeline.create();
+
+  public enum SideInputType {
+    ITERABLE,
+    MAP
+  }
+
+  /** Pipeline options for the test. */
+  public interface Options extends TestPipelineOptions {
+
+    @Description("The JSON representation of SyntheticBoundedInput.SourceOptions for main input.")
+    @Validation.Required
+    String getInputOptions();
+
+    void setInputOptions(String inputOptions);
+
+    @Description("The JSON representation of SyntheticBoundedInput.SourceOptions for side input.")
+    @Validation.Required
+    String getSideInputOptions();
+
+    void setSideInputOptions(String sideInputOptions);
+
+    @Description("The number of gbk to perform")
+    @Default.Integer(5)
+    Integer getFanout();
+
+    void setFanout(Integer shuffleFanout);
+
+    @Description("Type of side input")
+    @Default.Enum("ITERABLE")
+    SideInputType getSideInputType();
+
+    void setSideInputType(SideInputType sideInputType);
+  }
+
+  @BeforeClass
+  public static void setup() throws IOException {
+    PipelineOptionsFactory.register(Options.class);
+
+    options =
+        PipelineOptionsValidator.validate(
+            Options.class, TestPipeline.testingPipelineOptions().as(Options.class));
+
+    inputOptions = fromString(options.getInputOptions());
+    sideInputOptions = fromString(options.getSideInputOptions());
+  }
+
+  // TODO: Should this be a fanout test too?
+  @Test
+  public void sideInputLoadTest() {
+    PCollection<KV<byte[], byte[]>> input =
+        pipeline.apply("Read input", SyntheticBoundedIO.readFrom(inputOptions));
+
+    PCollection<KV<byte[], byte[]>> sideInputCollection =
+        pipeline.apply("Read side input", SyntheticBoundedIO.readFrom(sideInputOptions));
+
+    SideInputType sideInputType = options.getSideInputType();
+
+    if (sideInputType == ITERABLE) {
+      PCollectionView<Iterable<KV<byte[], byte[]>>> sideInput =
+          sideInputCollection.apply(View.asIterable());
+      input.apply(ParDo.of(new ConsumeIterable(sideInput)).withSideInputs(sideInput));
+    } else if (sideInputType == MAP) {
+
+      // For the sake of example completeness I used multimap here,
+      // because Map requires none key duplicates in the PCollection (otherwise fails).
+      // TODO: Should we GroupByKey or Combine.perKey() and use asMap() here?
+      PCollectionView<Map<byte[], Iterable<byte[]>>> sideInput = sideInputCollection
```

Comment:
Ok, acknowledged. I considered GBK -> view.asMap too, but it felt redundant in case multimap is ok. we can also add another "else if" branch in the test and test more use cases if this gives any value. WDYT?